scripts/upload_gatk_docs/upload_gatk_tools_docs.py

```diff
@@ -0,0 +1,238 @@
+                                 "sorting": "title",  # will sort contents alphabetically by article title
+                                 "locale": "en-us",
+                                 "name": gatk_version,
+                                 "description": "Tool documentation for GATK release " + gatk_version
+                                 }}
+
+        data = json.dumps(json_file)
+
+        try:
+            # create the folder
+            response = requests.post(url_Tool_Index, headers=headers, data=data, auth=(username + '/token', token))
+        except: 
+            exit("ERROR: Can't make section")
+
+        url = response.json()['section']['url']  # folder url
+
+    # edit url to be the API call to post an article inside
+    post_section_url = url.replace('.json', '/articles.json')
+
+    return post_section_url
+
+
+def post_articles(gatk_version, article_dict, username, token, headers, gatk_team_id, permission_group_id, user_id):
+    """Post articles to Zendesk."""
+    # get the post url (API call) for the GATK version section folder (e.g. '4.1.4.0') (create the folder if it doesn't exist yet)
+    post_section_url = post_section(gatk_version, username, token, headers)
+
+    # first loop thorugh and post all articles, collecting their urls
+    for article in article_dict.values():  # each `article` is a GatkDocs class
+        # post articles in that section folder
+        article.post(post_section_url, user_id, gatk_team_id, permission_group_id, headers, username, token)  # this sets article.url as well
+
+
+def update_articles(article_dict, gatkdoc_version, headers, username, token, updatefolder=None):
+    """Update articles with correct article links."""
+    # Creating the folder for the updated files
+    if updatefolder is None:
+        updatefolder = 'gatkdoc_update_' + gatkdoc_version
+    if not os.path.exists(updatefolder):
+        os.makedirs(updatefolder)
+
+    # Get each article
+    for article in article_dict.values():
+        print('in original article ' + article.file_name)
+        with open(article.local_path, 'r') as clean_file:
+            clean_file_data_soup = bs(clean_file, 'html.parser')
+
+        # Create Update file
+        update_file_path = os.path.join(updatefolder, article.file_name)
+        with open(update_file_path, 'w') as update_file:
+            # Getting all the links
+            for url in clean_file_data_soup.findAll('a'):
+                original_url = url.get('href')
+                if original_url in article_dict.keys():
+                    fixed_url_text = original_url.replace(original_url, article_dict[original_url].url)
+                    fixed_url = clean_file_data_soup.new_tag("a", href=fixed_url_text)
+                    fixed_url.string = article_dict[original_url].title
+                    url.replace_with(fixed_url)
+
+            # Format updated GATK Doc HTML
+            format_html = clean_file_data_soup.prettify()
```

Comment:
what kinds of formatting does `prettify` do? a comment here briefly explaining that would be helpful!